goon/browserassets/js/browserOutput.js

```diff
@@ -166,30 +166,79 @@ function emojiparse(el) {
 	}
 }
 
-// Colorizes the highlight spans
-function setHighlightColor(match) {
-	match.style.background = opts.highlightColor
+// Recolorizes the highlight spans
+function setHighlightColor() {
+	var highlightspans = document.getElementsByClassName("highlight")
+	for(var i in highlightspans){
+		highlightspans[i].setAttribute("style","background-color:"+opts.highlightColor)
+	}
+}
+
+function escapeRegex(input){ // put this in a function incase it ever needs to be used elsewhere, makes code cleaner
+	return input.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
 }
 
 //Highlights words based on user settings
 function highlightTerms(el) {
+
 	if(regexHasError) return; //just stop right there ig the regex is gonna except
+
+	function highlightRecursor(element,term){ //recursor function to do the highlighting proper
+		var regex = new RegExp("("+term+")","gi")
+
+		function replace(str) {
+			return str.replace(regex,'<span class="highlight" style="background-color:'+opts.highlightColor+'">$&</span>')
```

Comment:
Would the "highlight" span class's rules get overridden by the explicit color set here

Reply:
No, the highlight class will still follow the css rules, however inline styling will override  a stylesheet for that element, but nnot the rest of the class